### ISSUE TITLE
Reconstruct Bluesky PV connections

### ISSUE THREAD

**Issue by qzhang234** [aps-8id-dys/ipython-8idiuser#245]
The one and half week long Bluesky operation (10/01 - 10/12) has provided us with opportunities to make Bluesky fundamentally more stable and potentially even more error-resistant than Spec. 

Specifically, regarding the myriad of PV time-out issues that we have encountered over the past week, I would like to propose that we regroup all EPICS PVs that Bluesky interacts with into the four following groups:

A. PVs that do not change within one file header (e.g. q map, cycle name, most of StrReg);
B. PVs that change within one file header but is not essential to analysis (e.g. ring current, time of measurement);
C. PVs that change within one file header and may be essential to analysis (e.g. temperature, x/z motor position, .etc)
D. PVs that are prerequisites to measurements (A, D and I shutter).
 
During operation:
A: Read only once at the beginning of each _for_ loop. __Crash__ if an error occurs;
B. Read for each measurement. __Ignore and Proceed__ if an error occurs;
C. Read for each measurement. __Crash__ if an error occurs;
D. Read for each measurement. __Hang__ if conditions are not met; __Ignore and Proceed__ if an error occurs.

I believe that, before we start making try-catch-rethrow blocks to bypass various PV timeout and Rigaku hang issues (e.g. #233 #243, #244), it may be better to narrow down the list of PVs that we at 8-ID-I should really care about, as it significantly tightens the scope of the work and make the `exception` case much better defined. 

I also believe that, once all the PV connections have been reconstructed, the only exception that we really need to catch is the hang from Rigaku LabView interface. I will create a separate issue and invite Nakaye and Piotr to contribute to this, but it does not hurt to have a second solution to this problem as the chance of the bug occurring is less than 1/10,000 and there is no guarantee that switching to EPICS will fix it.

Any suggestions and comments are highly welcome.

**Comment by qzhang234** [comment]
@prjemian Also would it make sense for us to define a class called `ScanStatus`, which instantiates at the beginning of each scan and keeps track of the PV values for the most recent measurement, so that when the scan crashes we can immediately know which PV is the culprit?

**Comment by prjemian** [comment]
Let's leave this open for consideration but there are some problems within ophyd that should resolve the issues here AND also at 9-ID-C USAXS.  Since I believe the problem is fundamental to how ophyd use Channel Access monitors, let's work through https://github.com/bluesky/ophyd/pull/922 first, try it and come back here if the problems remain.
